### PR TITLE
Fix unused functions and variables

### PR DESCRIPTION
**Pull request by bz-liu**

### CODE REVIEW

File: src/IR/Builder.cpp [hunk 1]
```diff
@@ -24,11 +26,15 @@ limitations under the License.
...
-}
+
+bool SHOW_TODO_CASE = false;
+
+////bz: tmp comment off, since defined but not used
+// bool hasNoAliasMD(const llvm::Instruction *inst) {
+//   llvm::AAMDNodes AAMD;
+//   inst->getAAMetadata(AAMD);
+//   return AAMD.NoAlias != nullptr;
+// }
```
Comment (BradSwain) [inline src/IR/Builder.cpp:37]: This code can be removed.

File: src/IR/Builder.cpp [hunk 2]
```diff
@@ -84,10 +90,19 @@ FunctionSummary race::generateFunctionSummary(const llvm::Function &func) {
...
       } else if (auto branchInst = llvm::dyn_cast<llvm::BranchInst>(inst)) {
         // TODO: what should this do?
+        if (SHOW_TODO_CASE) {
+          std::cout << "TO HANDLE: " << branchInst << std::endl;
+        }
       } else if (auto switchInst = llvm::dyn_cast<llvm::SwitchInst>(inst)) {
         // TODO: what should this do?
+        if (SHOW_TODO_CASE) {
+          std::cout << "TO HANDLE: " << switchInst << std::endl;
+        }
```
Comment (BradSwain) [inline src/IR/Builder.cpp:105]: The empty if branches can be removed.

File: src/IR/Builder.cpp [hunk 3]
```diff
@@ -16,6 +16,8 @@ limitations under the License.
 #include <llvm/IR/Dominators.h>
 #include <llvm/IR/Instructions.h>
 
+#include <iostream>
+
```
Comment (BradSwain) [inline src/IR/Builder.cpp:20]: use `llvm::outs()` instead of `cout`

File: src/IR/IR.cpp [hunk 4]
```diff
@@ -52,6 +52,9 @@ llvm::raw_ostream &race::operator<<(llvm::raw_ostream &os, const IR::Type &type)
     case IR::Type::Call:
       os << "CALL";
       break;
+    default:
+      os << "OTHERS";
+      break;
```
Comment (BradSwain) [inline src/IR/IR.cpp:57]: The IR `operator<<` can be deleted.

**Review by BradSwain — CHANGES_REQUESTED** [review]

File: src/IR/Builder.cpp (hunk 2 above)
Comment (bz-liu) [reply src/IR/Builder.cpp:105]: I'm not sure this can be removed or not, there are 'TODO's.

File: src/IR/IR.cpp (hunk 4 above)
Comment (bz-liu) [reply src/IR/IR.cpp]: removed

File: src/IR/Builder.cpp (hunk 2 above)
Comment (bz-liu) [reply src/IR/Builder.cpp]: removed

File: src/IR/Builder.cpp (hunk 1 above)
Comment (bz-liu) [reply src/IR/Builder.cpp]: removed

File: src/IR/Builder.cpp (hunk 3 above)
Comment (bz-liu) [reply src/IR/Builder.cpp]: removed if we do not need the empty branches

**Review by BradSwain — APPROVED** [review]

**Pull request merged**